Sources/Datadog/Core/FeaturesConfiguration.swift

```diff
@@ -140,9 +140,13 @@ extension FeaturesConfiguration {
 }
 
 private func ifValid(environment: String) throws -> String {
-    let regex = #"^[a-zA-Z0-9_]+$"#
+    /// 1. starts with a letter
```

Comment:
As @mariusc83 concluded in https://github.com/DataDog/dd-sdk-android/pull/369#discussion_r486145119 this first requirement doesn't apply here, as what we validate here is the value of the tag. It's name will be `env:` (added by the backend), so it will always start with the letter `e`. The regexp should be then:
```swift
[a-zA-Z0-9_:./-]{0,195}[a-zA-Z0-9_./-]
```